public/app/plugins/panel/alertlist/UnifiedAlertList.tsx

```diff
@@ -127,14 +128,20 @@ export function UnifiedAlertList(props: PanelProps<UnifiedAlertListOptions>) {
   );
 }
 
-function sortRules(sortOrder: SortOrder, rules: PromRuleWithLocation[]) {
+function sortRules(sortOrder: SortOrder, rules: CombinedRuleWithLocation[]) {
   if (sortOrder === SortOrder.Importance) {
     // @ts-ignore
     return sortBy(rules, (rule) => alertDef.alertStateSortScore[rule.state]);
   } else if (sortOrder === SortOrder.TimeAsc) {
-    return sortBy(rules, (rule) => getFirstActiveAt(rule.rule) || new Date());
+    return sortBy(rules, (rule) => {
+      const alertingRule: AlertingRule | undefined = getAlertingRule(rule) ?? undefined;
```

Comment:
What happens if the rule is not alerting rule? Can we display anything valuable in that case?

Reply:
at this point this list is filtered with only alerting rules, but i need to check this because of TypeScript 🙂

Reply:
Ok :) Could you add a comment that this is only for Typescript?